src/structurizr/view/view_set.py

```diff
@@ -75,25 +77,85 @@ class ViewSet(ModelRefMixin, AbstractBase):
+        component_views: Iterable[ComponentView] = (),
         configuration: Optional[Configuration] = None,
         **kwargs
     ) -> None:
         """Initialize a view set."""
         super().__init__(**kwargs)
         # self.enterprise_context_views = set(enterprise_context_views)
-        self.system_landscape_views = set(system_landscape_views)
-        self.system_context_views = set(system_context_views)
-        self.container_views = set(container_views)
+        self.system_landscape_views: Iterable[SystemLandscapeView] = set(
+            system_landscape_views
+        )
+        self.system_context_views: Iterable[SystemContextView] = set(
+            system_context_views
+        )
+        self.container_views: Iterable[ContainerView] = set(container_views)
+        self.component_views: Iterable[ComponentView] = set(component_views)
         # TODO
         self.configuration = Configuration() if configuration is None else configuration
         self.set_model(model)
 
+    @classmethod
+    def hydrate(cls, views: ViewSetIO, model: "Model") -> "ViewSet":
+        system_landscape_views = []
+        for view_io in views.system_landscape_views:
+            view = SystemLandscapeView.hydrate(view_io, model=model)
+            cls.hydrate_view(view, model=model)
+            system_landscape_views.append(view)
+
+        system_context_views = []
+        for view_io in views.system_context_views:
+            software_system = model.get_software_system_with_id(
+                view_io.software_system_id
+            )
+            view = SystemContextView.hydrate(view_io, software_system=software_system)
+            cls.hydrate_view(view, model=model)
+            system_context_views.append(view)
+
+        container_views = []
+        for view_io in views.container_views:
+            software_system = model.get_software_system_with_id(
+                id=view_io.software_system_id,
+            )
+            view = ContainerView.hydrate(view_io, software_system=software_system)
+            cls.hydrate_view(view, model=model)
+            container_views.append(view)
+
+        component_views = []
+        for view_io in views.component_views:
+            container = model.get_element(view_io.container_id)
+            view = ComponentView.hydrate(view_io, container=container)
+            cls.hydrate_view(view, model=model)
+            component_views.append(view)
+
+        return cls(
+            model=model,
+            # TODO:
+            # enterprise_context_views: Iterable[EnterpriseContextView] = (),
+            system_landscape_views=system_landscape_views,
+            system_context_views=system_context_views,
+            container_views=container_views,
+            component_views=component_views,
+            configuration=Configuration.hydrate(views.configuration),
+        )
+
+    @classmethod
+    def hydrate_view(cls, view: View, model: "Model") -> None:
```

Comment:
I'd hide this method from the public interface. Maybe the work done in this model should be done when adding an element to the `ViewSet`?

```suggestion
    def _hydrate_view(cls, view: View, model: "Model") -> None:
```